ViewStore.md

```diff
@@ -0,0 +1,135 @@
+
+## Performing `Action`s
+
+To perform an action, we typically create a `PassthroughSubject`, which is then used in the `Combine` pipeline to update the `viewState`. We call the `send(_ action:)` API to perform the action, and then typically send a new value to the `PassthroughSubject`.
+
+```swift
+enum Action {
+    case toggleShowsPhotoCount(Bool)
+}
+
+private let showsPhotosCountPublisher = PassthroughSubject<Bool, Never>()
+
+func send(_ action: Action) {
+    switch action {
+    case let .toggleShowsPhotoCount(showsPhotoCount):
+        showsPhotosCountPublisher.send(showsPhotoCount)
+    }
+}
+```
+
+Additionally, `PassthroughSubject`s are prepended with an initial value, since their usage in `combineLatest` requires that a value be emitted before the `combineLatest` can emit.
+
+```swift
+let showsPhotosCountPublisher = self.showsPhotosCountPublisher.prepend(ViewState.initial.showsPhotoCount)
+photoPublisher
+    .combineLatest(showsPhotosCountPublisher)
+    .map { /* transformation to ViewState */ }
+    .assign(to: &$viewState)
+```
+
+## Bindings
+
+Many SwiftUI APIs accept bindings for state that is both read and written to. Binding properties or methods are frequently declared on view stores as a convenience for working with these APIs. To keep a single source of truth, the binding typically reads a property on the `viewState` and performs an action that results in an update to the view state.
+
+```swift
+var showsPhotoCount: Binding<Bool> {
+    return Binding<Bool> {
+        self.viewState.showsPhotoCount
+    } set: { newValue in
+        self.send(.toggleShowsPhotoCount(newValue))
+    }
+}
+```
+
+Rather than using `send(_ action:)` directly, the `View` would instead use this binding.
+
+```swift
+Toggle("Show Count", isOn: store.showsPhotoCount)
+```
+
+As a convenience, a [`makeBinding` API](https://github.com/Lickability/ViewStore/blob/main/Sources/ViewStore/ViewStore%2BBindingAdditions.swift) is provided in an extension of the `ViewStore` protocol to create a succinct syntax for this common case. This extension uses [CasePaths](https://github.com/pointfreeco/swift-case-paths).
+
+```swift
+var showsPhotoCount: Binding<Bool> {
+    makeBinding(viewStateKeyPath: \.showsPhotoCount, actionCasePath: /Action.toggleShowsPhotoCount)
```

Comment:
This isn't really feedback and probably is just an exposure thing but for whatever, this feels harder to understand at the call site than just the binding init to me.

Reply:
I definitely think it is harder to understand, and that’s fair feedback. It’s an unusual and unfamiliar syntax. We introduce it _second_ for that reason. I see it as useful for saving lines of code / boilerplate, especially in a more complex view store with many bindings, but its usage is definitely optional.